### DISCUSSION TITLE
Suggested task for Outreachy Summer 22: 1

### DISCUSSION THREAD

**Discussion by pgleeson** [ModECI/MDF#224] [Ideas]
A suggested set of tasks for applicants to the Outreachy 2022 call.

1) Create your own fork of the main MDF repository and clone it locally.
2) Change to the development branch and install this version of MDF on your machine 
3) Run the notebook at https://github.com/ModECI/MDF/blob/development/examples/SimpleExample.ipynb on your machine.
4) Add a new section to the notebook to modify one of the nodes in the existing graph to add a state variable, and then run a simulation with it, plotting the result in the notebook & explaining the steps along the way. You can use this example as a basis, but use a different expression for the time derivative: https://github.com/ModECI/MDF/tree/development/examples/MDF#states.
5) Don’t open a pull request, but mention @pgleeson and @sanjayankur31 in a commit with your updated notebook and we’ll have a look.

**Comment by DeejaKamran** [comment]
Heyy @pgleeson 
Thank you for your task. I have run the SimpleExample.ipynb file and now trying to get a better understanding of MDF from https://github.com/ModECI/MDF#readme. Is there any other material you recommend?

**Reply by pgleeson** [reply]
Unfortunately the documentation is a bit limited at the moment. There is this: https://mdf.readthedocs.io, but it's mainly API docs generated from the code. The examples which have been developed try to demonstrate the core functionality present.

**Comment by amitha6** [comment]
Hey @pgleeson 
I followed all the above steps and ran the simpleExample notebook. Its giving the following error.
ExecutableNotFound: failed to execute ['dot', '-Kdot', '-Tpng', '-O', 'AB'], make sure the Graphviz executables are on your systems' PATH
Need help.!!

**Reply by pgleeson** [reply]
Hi @amitha6, you need to install graphviz and make sure that you can run it on your system, i.e. the executable `dot` can be run.

**Reply by esraa-abdelmaksoud** [reply]
@amitha6 I have faced the same problem and here's how you can solve it. If you're sure that graphviz is already installed and the path is in the environment variables, you will notice in the error details that one of graphviz files is missing. This file is required for inline graph visualization in Windows. Use the following [link](https://graphviz.org/download/#windows) to download your corresponding package and remember to tick "Add path to environment variables". Then, restart your system and it should work for you because the missing file will be added.

**Comment by 29riyasaxena** [comment]
Hey @pgleeson @sanjayankur31!
I made a commit, I have mentioned you both in them. Is it visible to you?

**Comment by pgleeson** [comment]
Please remember for anyone carrying out this task, the project we'd like you to carry out during the internship is related to making educational notebooks for users to teach them about MDF and the concepts around it.

What we're looking for in this task is more than the ability to add a state in the notebook and get it to run, it's more about assessing how well you can explain technical details in a simple, step by step way and make it into a well written, clean notebook.

Remember when you build on the existing notebook, to introduce the new concepts clearly, say why they are important, link to any other documentation, keep the example simple, even remove previous parts if necessary, and stick to the formatting/layout/style of the previous steps in the tutorial.

**Reply by Megha-Bose** [reply]
Hey @pgleeson,

I made a commit for this task. It would be great if you could provide some feedback on that.

Thanks in advance

**Comment by 29riyasaxena** [comment]
Hi! I have completed this task few days back, It would be great if you provide the feedback so that I can enter my contribution in the same. @pgleeson @sanjayankur31
